newton/examples/__init__.py

```
@@ -45,8 +162,36 @@ def run(example):
+        if not hasattr(example, "test"):
+            raise NotImplementedError("Example does not have a test method")
+        example.test()
+
     example.viewer.close()
 
+    if args is not None and args.test:
+        # generic tests for finiteness of Newton objects
+        if hasattr(example, "state_0"):
+            nan_members = find_nan_members(example.state_0)
+            if nan_members:
+                raise ValueError(f"NaN members found in state_0: {nan_members}")
```

Ruff (0.13.2)

Avoid specifying long messages outside the exception class

(TRY003)